app/jobs/compute_potential_duplicates.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+class ComputePotentialDuplicates < ApplicationJob
+  def perform(job)
+    persons_cache = Person.select(:id, :wca_id, :name, :dob, :country_id)
+    new_comers = job.competition
+                    .registrations
+                    .includes(:user)
+                    .select { it.wcif_status == "accepted" }
+                    .map(&:user)
+                    .select { it.wca_id.nil? }
```

Comment:
This should be feasible through one or two clever `has_many` relation on competition. Something along the lines of `has_many :accepted_registrations, -> { competing_status_accepted? }, class_name: "Registration"` and `has_many :accepted_newcomers, -> { where(wca_id: nil) }, class_name: "User", through: :accepted_registrations`

Reply:
Beware, I wrote this completely freestyle on GitHub without checking the actual code. Probably need to polish / fix the details

Reply:
That's really clever, thanks for that. Done. (made some minor changes)